backend/src/ledgers.ts

```diff
@@ -145,10 +237,9 @@ export async function updateCache(
   });
   cachedStats.sort(dateSorter);
 
-  // only store latest 30 days
   await redisClient.set(
     ledgersKey,
-    JSON.stringify(cachedStats.slice(0, LEDGER_DAY_COUNT)),
+    JSON.stringify(cachedStats.slice(0, LEDGER_ITEM_LIMIT[interval])),
   );
   await redisClient.set(pagingTokenKey, pagingToken);
 
```

Comment:
we can remove some of the code w/in `dateSorter` since the object keys are using timestamps now. Some of that logic was there because of edge cases of comparing dates w/o the year when they were in the format like:
```
[
  {
    "date": "06-13",
    "transaction_count": 4894311,
    "operation_count": 7576618
  },
  {
    "date": "06-12",
    "transaction_count": 5055858,
    "operation_count": 10093060
  },
  {
    "date": "06-11",
    "transaction_count": 5113215,
    "operation_count": 9201338
  },
...
```

The dateSorter function can just be this now, thanks to the timestamp key idea you guys had 😄 :
```
function dateSorter(a: LedgerStat, b: LedgerStat) {
  return new Date(b.date).getTime() - new Date(a.date).getTime();
}

```

Reply:
True, good catch!